backend/app.py

```diff
@@ -308,6 +314,138 @@ def create_machine():
+    )[0]
+    change_message = "Changed"
+
+    # Case 1: status was changed:
+    if status != existing_machine_infos["properties"]["machine_status"]:
+        updated_machine_entry["properties"]["machine_status"] = status
+        change_message += " status,"
+
+    # Case 2: if area was changed -> match to available areas
+    if area != existing_machine_infos["properties"]["area"]:
+        # Identify area
+        area, score = fuzzysearch.extract(area, COUNTRIES, limit=1)[0]
+        if score < 90:
+            return (
+                jsonify(
+                    {
+                        "error": "Could not match country. Provide country or US state name in English"
+                    }
+                ),
+                400,
+            )
+        updated_machine_entry["properties"]["area"] = area
+        change_message += " area,"
+
+    # Case 3: Title changed
+    if title != existing_machine_infos["properties"]["name"]:
+        updated_machine_entry["properties"]["name"] = title
+        change_message += " title,"
+
+    # Case 4: multimachine changed
+    try:
+        multimachine = int(request.args.get("multimachine"))
+    except ValueError:
+        # just put the multimachine as a string, we need to correct it then
+        multimachine = "TODO" + str(request.args.get("multimachine"))
+    if multimachine != 1:
+        updated_machine_entry["properties"]["multimachine"] = multimachine
+        change_message += " multimachine,"
+
+    # Case 5: paywall reported
+    paywall_new = True if request.args.get("paywall") == "true" else False
+    paywall_old = existing_machine_infos["properties"].get("paywall", False)
+    if paywall_new != paywall_old:
+        updated_machine_entry["properties"]["paywall"] = paywall_new
+        change_message += " paywall,"
+
+    # Case 6: address and / or location changed --> check for their correspondence
+    (lng_old, lat_old) = existing_machine_infos["geometry"]["coordinates"]
+    old_address = existing_machine_infos["properties"]["address"]
+    # if address or coordinates were changed, compare them and return warning if needed
+    address_okay = True
+    if latitude != lat_old or longitude != lng_old or address != old_address:
+        # Verify that address matches coordinates
+        found_coords, (lat, lng) = verify_machine_location(address, area, title)
+        # if address was changed but is not found (error only if address was changed)
+        if (not found_coords) and address != old_address:
+            return jsonify({"error": "Google Maps does not know this address"}), 400
+
+        dist = haversine((lat, lng), (latitude, longitude))
+        if dist > 1:  # km
+            address_okay = False  # triggers warning
+
+        # adapt dictionary entries
+        updated_machine_entry["properties"]["address"] = address
+        updated_machine_entry["properties"]["latitude"] = str(latitude)
+        updated_machine_entry["properties"]["longitude"] = str(longitude)
```

Comment:
I thought about using the found coordinates (lat, lng) here if only the address was changed by the user. I decided against it to give users the option to change the address without triggering a moving of the pin.

Reply:
I agree since this could result in unexpected behavior from the user side